packages/ckeditor5-link/src/linkcommand.ts

```diff
@@ -277,4 +288,37 @@ export default class LinkCommand extends Command {
 
 		return true;
 	}
+
+	private _updateLinkContent( model: Model, writer: Writer, range: Range, href: string ): Range {
+		const attributes = toMap( {} );
+
+		attributes.set( 'linkHref', href );
+
+		const text = writer.createText( href, attributes );
+
+		return model.insertContent( text, range );
+	}
+}
+
+function getTextFromRange( range: Range | null ): string {
+	if ( !range || range && !Array.from( range.getItems() ).length ) {
+		return '';
+	}
+
+	const firstNode = Array.from( range.getItems() )[ 0 ];
+
+	if ( firstNode.is( '$text' ) || firstNode.is( '$textProxy' ) ) {
+		return firstNode.data;
+	}
+
+	return '';
```

Comment:
I'd return `null` in this case. If there's no range or the range is empty or it has no text nodes, there's no text inside (semantics).